### PR TITLE
Fix ObjectPropertyDescriptor

### PR DESCRIPTION
**Pull request by hs0225**
Signed-off-by: Hosung Kim hs852.kim@samsung.com

### CODE REVIEW

File: src/runtime/Object.h [hunk 1]
```diff
@@ -509,13 +509,17 @@ class ObjectPropertyDescriptor {
 
     bool hasJSGetter() const
     {
-        ASSERT(!isDataProperty());
+        if (!isDataProperty()) {
+            return false;
```
Comment (clover2123) [inline src/runtime/Object.h:513]: It seems that this condition is reversed.
If property is data property type then return false value.
One more thing, would you please add `UNLIKELY` macro to this if statement?

File: src/runtime/Object.h [hunk 2]
```diff
@@ -509,13 +509,17 @@ class ObjectPropertyDescriptor {
...
+        }
         return m_getterSetter.hasGetter();
     }
 
     bool hasJSSetter() const
     {
-        ASSERT(!isDataProperty());
+        if (!isDataProperty()) {
+            return false;
```
Comment (clover2123) [inline src/runtime/Object.h:521]: Same at here too.

File: src/runtime/Object.h (hunk 1 above)
Comment (ksh8281) [reply src/runtime/Object.h:513]: IMO you can check this things on EscargotPublic.cpp for public API

File: src/api/EscargotPublic.cpp [hunk 3]
```diff
@@ -1150,12 +1150,12 @@ bool ObjectPropertyDescriptorRef::hasEnumerable() const
 
 void ObjectPropertyDescriptorRef::setConfigurable(bool configurable)
 {
-    ((ObjectPropertyDescriptor*)m_privateData)->setEnumerable(configurable);
+    ((ObjectPropertyDescriptor*)m_privateData)->setConfigurable(configurable);
```
Comment (ksh8281) [inline src/api/EscargotPublic.cpp:1153]: Could you add super simple test into to testapi.cpp?

**Comment by hs0225** [comment]
@clover2123 @ksh8281 I updated my PR. Please check again.

**Review by clover2123 — APPROVED** [review]
LGTM

**Review by ksh8281 — APPROVED** [review]

**Pull request merged**